### PR TITLE
VIR-50: Vacancy fill percentage validation

### PR DESCRIPTION
**Pull request by Peskarhu**
* Added validation logic to check that vacancy fill does not exceed vacancy size

### CODE REVIEW

**Comment by Peskarhu** [comment]
Made a small test on my end to try updating the vacancyFill > vacancySize.

before:

Näyttökuva 2024-10-25 003655

and trying to update:

Näyttökuva 2024-10-25 003734 

Tips for testing or better setup! :(

File: backend/src/Functions/Positions/CreatePosition.cs [hunk 1]
```diff
@@ -27,6 +27,12 @@ public async Task<IActionResult> Run(
 
         var (error, requestPosition) = await TryDeserializeRequestBody<Position>(req);
 
+        // Check that the fill percentage is not greater than the vacancy percentage
+        if (requestPosition.VacancyFill > requestPosition.VacancySize)
```
Comment (0x464e) [inline backend/src/Functions/Positions/CreatePosition.cs:34]: Validations need to be after we check if the deserialization was successful (`if (error is not null)`).  
Here it is not guaranteed that `requestPosition` is non-null.

**Review by 0x464e — CHANGES_REQUESTED** [review]

**Comment by 0x464e** [comment]
> Tips for testing or better setup! :(

You can use your IDE's built in HTTP client.  
For VS Code, I believe you want [this](https://marketplace.visualstudio.com/items?itemName=humao.rest-client) plugin.

For Rider, or any other Jetbrains IDE, a HTTP client is built-in. I use that and wrote a tip about it on an earlier PR:  
https://github.com/virkarekisteri/virkarekisteri/pull/7#issuecomment-2411897003

Perhaps even better would be to setup Postman

File: backend/src/Functions/Positions/CreatePosition.cs (hunk 1 above)
Comment (Peskarhu) [reply backend/src/Functions/Positions/CreatePosition.cs:34]: True, didnt notice late at night. Will update this soon

**Review by 0x464e — APPROVED** [review]

**Pull request merged**